### PR TITLE
Reorganize native realm dependencies into package

### LINKED ISSUE #5584: Clean up the root of the repository
* Remove `docs`, `lib`,  `src`, `CMakeLists.txt` and `types`
* Migrate the `platform.cpp`/`platform.mm` to the correct package

### PR DESCRIPTION
**Pull request by takameyer**
## What, How & Why?
Prep for build optimization.  Move any necessary build files and scripts into `packages/realm`.  Remove any unnecessary build files as they are found.

This closes #5584

Goals for the PR, if there is anything else that could be removed, please enlighten me. 

## ☑️ ToDos

* [ ] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [ ] 🚦 Tests
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

### CODE REVIEW

**Review by kneth — APPROVED** [review]
Merge it when CI is 🟢

File: packages/realm/scripts/build-android.ts [hunk 1]
```diff
@@ -15,19 +15,25 @@
...
 
 const NDK_VERSION = "25.1.8937393";
 
 const { ANDROID_SDK_ROOT } = process.env;
+
+if (!ANDROID_SDK_ROOT) {
+  console.error(`Missing env variable ANDROID_SDK_ROOT`);
+  process.exit(1);
+}
+
```
Comment (kraenhansen) [inline packages/realm/scripts/build-android.ts:36]: Now I understand why you wanted to hold off the merge of #6402 😬

**Review by kraenhansen — APPROVED** [review]
Let's go!

File: packages/realm/scripts/build-ios.sh [hunk 2]
```diff
@@ -97,15 +98,15 @@ DEVELOPER_DIR="${DEVELOPER_DIR:-${SELECTED_DEVELOPER_DIR}}"
...
     -DCMAKE_ARCHIVE_OUTPUT_DIRECTORY="$(pwd)/out/$<CONFIG>\$EFFECTIVE_PLATFORM_NAME" \
 
 DEVELOPER_DIR="$DEVELOPER_DIR" xcodebuild build \
     -scheme realm-js-ios \
     "${DESTINATIONS[@]}" \
     -configuration $CONFIGURATION \
-    CC="$PROJECT_ROOT/scripts/ccache-clang.sh" \
-    CXX="$PROJECT_ROOT/scripts/ccache-clang++.sh" \
+    CC="$PROJECT_ROOT/../../scripts/ccache-clang.sh" \
+    CXX="$PROJECT_ROOT/../../scripts/ccache-clang++.sh" \
```
Comment (kraenhansen) [inline packages/realm/scripts/build-ios.sh:109]:
```suggestion
    CC="$PROJECT_ROOT/scripts/ccache-clang.sh" \
    CXX="$PROJECT_ROOT/scripts/ccache-clang++.sh" \
```

I don't think these needs an update?

File: packages/realm/scripts/build-android.ts (hunk 1 above)
Comment (takameyer) [reply packages/realm/scripts/build-android.ts:36]: Yeah, but it should be easily merged.

File: packages/realm/scripts/build-ios.sh (hunk 2 above)
Comment (takameyer) [reply packages/realm/scripts/build-ios.sh:109]: It does need this.  I'll just merge it in its working form right now.  The build optimizations for iOS will overwrite this anyway.

**Pull request merged**